domain/domain.go

```diff
@@ -535,7 +535,8 @@ func NewDomain(store kv.Storage, ddlLease time.Duration, statsLease time.Duratio
 
 // ResetHandle resets the domain's infoschema handle. It is used for testing.
 func (do *Domain) ResetHandle(store kv.Storage) {
-	do.infoHandle = infoschema.NewHandle(store)
+	// `do.infoHandle` may refer to being accessed concurrently, so atomic operations are performed here.
+	atomic.StorePointer((*unsafe.Pointer)(unsafe.Pointer(&do.infoHandle)), unsafe.Pointer(infoschema.NewHandle(store)))
```

Comment:
the data race is between write and read.
but, there is only use atomic.Store but not use atomic.Get, Is it worked?

Reply:
No, need to protect reading at do.infoHandle.

Reply:
@winkyao @crazycs520 Done